kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DriverDefEditorServiceTest.java

```
@@ -0,0 +1,169 @@
+            uri = resource.toURI();
+            when( artifactResolver.resolve( driverDef.getGroupId(), driverDef.getArtifactId(), driverDef.getVersion() ) ).thenReturn( uri );
+        } catch ( Exception e ) {
+        }
+    }
+
+    @Override
+    protected DefEditorContent getExpectedContent() {
+        return driverDefEditorContent;
+    }
+
+    @Override
+    protected String getExpectedDefString() {
+        return DriverDefSerializer.serialize( driverDef );
+    }
+
+    @Override
+    protected String getExpectedFileName() {
+        return driverDef.getName() + ".driver";
+    }
+
+    @Override
+    protected Def getExpectedDef() {
+        return driverDef;
+    }
+
+    @Override
+    protected Def getOriginalDef() {
+        return originalDriverDef;
+    }
+
+    @Override
+    protected String getOriginalDefString() {
+        return DriverDefSerializer.serialize( originalDriverDef );
+    }
+
+    @Override
+    protected void verifyCreateConditions( boolean global ) {
+        //we wants that:
+        try {
+            // 2) the definition was deployed.
+            verify( runtimeManager, times( 1 ) ).deployDriver( driverDef, DeploymentOptions.create() );
+        } catch ( Exception e ) {
+        }
+        // 3) the notification was fired.
+        NewDriverEvent expectedEvent;
+        if ( global ) {
+            expectedEvent = new NewDriverEvent( driverDef, SESSION_ID, IDENTITY );
+        } else {
+            expectedEvent = new NewDriverEvent( driverDef, project, SESSION_ID, IDENTITY );
+        }
+        verify( newDriverEvent, times( 1 ) ).fire( expectedEvent );
+    }
+
+    @Override
+    protected void verifySaveConditions() {
+        //we wants that
+        try {
+            // 3) the definition was deployed
+            verify( runtimeManager, times( 1 ) ).deployDriver( driverDef, DeploymentOptions.create() );
+        } catch ( Exception e ) {
```

Same comment about swallowing the exception.